tests/e2e/tests/regression/browser/filtering.e2e.ts

```diff
@@ -206,33 +206,41 @@ test
     })
     .after(async() => {
         // Delete database
-        await browserPage.deleteKeyByName(keyName);
         await deleteStandaloneDatabaseApi(ossStandaloneBigConfig);
     })('Verify that user can filter per key name using patterns in DB with 10-50 millions of keys', async t => {
-        // Create new key
-        keyName = `KeyForSearch-${chance.word({ length: 10 })}`;
-        await browserPage.addSetKey(keyName);
-        // Search by key name
+        keyName = 'device*';
         await browserPage.selectFilterGroupType(KeyTypesTexts.Set);
-        // Verify that required key is displayed
-        await browserPage.searchByKeyNameWithScanMore('KeyForSearch*', keyName);
-        // Verify that required key is displayed in tree view
+        await browserPage.searchByKeyName(keyName);
+        for (let i = 0; i < 10; i++) {
+            // Verify that keys are filtered
+            await t.expect(await browserPage.keyNameInTheList.nth(i).textContent).contains('device', 'Keys filtered incorrectly by key name')
+                .expect(await browserPage.keyNameInTheList.nth(i).textContent).contains('set', 'Keys filtered incorrectly by key type');
+        }
         await t.click(browserPage.treeViewButton);
-        await browserPage.searchByKeyNameWithScanMore('KeyForSearch*', keyName);
+        //Verify that user can use the "Scan More" button to search per another 10000 keys
+        for (let i = 10; i < 100; i += 10) {
+            // scannedValue = scannedValue + 10;
+            await t.expect(browserPage.progressKeyList.exists).notOk('Progress Bar is not displayed', { timeout: 30000 });
+            const scannedValueText = await browserPage.scannedValue.textContent;
+            const regExp = new RegExp(`${i} 00` + '.');
+            await t.expect(scannedValueText).match(regExp, `The database is automatically scanned by ${i} 000 keys`);
+            await t.doubleClick(browserPage.scanMoreButton);
+            await t.expect(browserPage.progressKeyList.exists).ok('Progress Bar is displayed', { timeout: 30000 });
+        }
```

Comment:
As I remember, this part of code was in some other e2e. Can you please create a function and reuse it?
And also please add validation for number of found keys that should be increased

Reply:
updated, thanks!